wrappers/python/simtk/openmm/app/charmmpsffile.py

```diff
@@ -466,6 +467,37 @@ def __init__(self, psf_name, periodicBoxVectors=None, unitCellDimensions=None):
         else:
             self.box_vectors = periodicBoxVectors
 
+    def _build_exclusion_list(self):
+        pair_12_set = set()
+        pair_13_set = set()
+        pair_14_set = set()
+        for bond in self.bond_list:
+            a1, a2 = bond.atom1, bond.atom2
+            pair = (min(a1.idx, a2.idx), max(a1.idx, a2.idx),)
+            pair_12_set.add(pair)
+        for bond in self.bond_list:
+            a2, a3 = bond.atom1, bond.atom2
+            for a1 in a2.bond_partners:
+                pair = (min(a1.idx, a3.idx), max(a1.idx, a3.idx),)
+                if a1 != a3:
+                    pair_13_set.add(pair)
+            for a4 in a3.bond_partners:
+                pair = (min(a2.idx, a4.idx), max(a2.idx, a4.idx),)
+                if a2 != a4:
+                    pair_13_set.add(pair)
```

Comment:
Is this loop actually needed?  Shouldn't the loop above already have found all triples?

Reply:
Yes, it is needed. For a five-member ring, there are five bonds: (1-2, 2-3, 3-4, 4-5, 1-5). If it only considers a1-a2-a3 but not a2-a3-a4, then the pair_13_set will be (5-1-2, 1-2-3, 2-3-4, 3-4-5, 2-1-5), so 4-5-1 is missed.

Reply:
Ok, just wanted to make sure!